### PR TITLE
[script.timers] 2.0.0

### PR DESCRIPTION
**Pull request by Heckie75**
### Description
Powerful timers for KODI

I wanted to have a Kodi addon that is a sleep timer, a doze timer, has several other timer slots and combines it with fading volume in and out over a certain period.

Overview of features:
* 10 timer slots. All of them can be quickly set up by using context menu
* 2 additional slots for sleep and doze timers with single-click-setup 
* Timers can play any ressource that it available in Kodi, e.g. music, video files, TV/radio programs, ressources from 3rd party plugins, e.g. Zattoo channels. 
* Different schedule modes: once, everyday, Mon-Fri, Fri-Sat, Sat-Sun, Sun-Thu, Mon-Thu, specific weekday and many more
* Date change is supported, e.g. from 23:30 (p.m.) until 1:30 (a.m.) 
* Two end modes, i.e. duration or specific time
* Actions related to media: start media and stop at end, just start media, start media at end, stop media immediately, stop media at end, powerdown system
* Linear fading in timer period: fade-in, fade-out, no fading. Min and max volume can be set for each timer
* Custom label for timer
* After KODI startup timers, that are in period, start retroactivly altought KODI was not running at start time. Fading volume is calculated correctly.
* MS Windows only: Feature in order to prevent that Windows displays lock screen if Kodi idles

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

### CODE REVIEW

**Review by ronie — CHANGES_REQUESTED** [review]
please remove the `script.timers/__init__.py` file, as it's not needed.

File: script.timers/addon.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# see https://www.raspberrypi.org/forums/viewtopic.php?t=166912
+import _strptime
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcvfs
+
+from resources.lib.timer.scheduler import CHECK_INTERVAL, Scheduler
+
+__PLUGIN_ID__ = "script.timers"
```
[the remark this answers is not in the document]
Comment (Heckie75) [reply script.timers/addon.py:15]: Ok, has been changed

File: script.timers/addon.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			<screenshot>resources/assets/screenshot_03.png</screenshot>
+			<screenshot>resources/assets/screenshot_04.png</screenshot>
+			<screenshot>resources/assets/screenshot_05.png</screenshot>
+			<screenshot>resources/assets/screenshot_06.png</screenshot>
+			<screenshot>resources/assets/screenshot_07.png</screenshot>
+			<screenshot>resources/assets/screenshot_08.png</screenshot>
+			<screenshot>resources/assets/screenshot_09.png</screenshot>
+			<screenshot>resources/assets/screenshot_10.png</screenshot>
+			<screenshot>resources/assets/screenshot_11.png</screenshot>
+			<screenshot>resources/assets/screenshot_12.png</screenshot>
```
[the remark this answers is not in the document]
Comment (Heckie75) [reply script.timers/addon.xml:75]: Ok, changed as well

File: script.timers/program.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import xbmc
+import xbmcaddon
+
+from addon import __PLUGIN_ID__
+
+addon = xbmcaddon.Addon(id=__PLUGIN_ID__)
+
```
[the remark this answers is not in the document]
Comment (Heckie75) [reply script.timers/program.py:7]: Also changed

File: script.timers/resources/lib/timer/abstract_set_timer.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+
+class AbstractSetTimer:
+
+    addon = None
+
+    def __init__(self, listitem):
+
+        self.addon = xbmcaddon.Addon(id=__PLUGIN_ID__)
+
```
[the remark this answers is not in the document]
Comment (Heckie75) [reply script.timers/resources/lib/timer/abstract_set_timer.py:27]: Also changed

[the remark this answers is not in the document]
Comment (Heckie75) [reply script.timers/resources/lib/timer/abstract_set_timer.py:26]: I have just pushed the changes. Pls. take another look at it.

Thanks!

**Review by ronie — APPROVED** [review]

**Comment by ronie** [comment]
thanx for the quick fixes!

looks ok to me now, apart from 1 thing i missed when reviewing the addon.
please add the python dependency to your addon.xml file:
```
	<requires>
		<import addon="xbmc.python" version="3.0.0"/>
	</requires>
```

once that's in place, this pr should be good to go.

**Comment by Heckie75** [comment]
I have also added the <requires>. I hope that it is fine. 

Pls. take a look at it.

Thanks for your efforts.

**Comment by ronie** [comment]
thx!

**Pull request merged**